File: StorageManagement/EletricGo/EletricGo/Domain/Storages/Location.cs
using EletricGo.Domain.Shared;

namespace EletricGo.Domain.Storages
{
    public class Location : Entity<LocationId>
    {
        public string Latitude { get; set; }
        public string Longitude { get; set; }
        public string Altitude { get; set; }
        public Address? Address { get; set; }


        public Location() {
        }

        public Location(string latitude, string longitude, string altitude, Address addr)
        {
            Id = new LocationId(Guid.NewGuid());
            Latitude = latitude;
            Longitude = longitude;
            Altitude = altitude;
            if(addr != null) 
                Address = new Address(addr.Street, addr.Door, addr.Floor, addr.PostalCode, addr.City); 
        }
    }
}

File: code/spa/src/models/location.ts
import type Address from "./address";

export default class Location {
    Latitude: string;
    Longitude: string;
    Altitude: string;
    Address: Address;

    constructor(latitude: string, longitude: string, altitude: string, address: Address) {
        this.Latitude = latitude;
        this.Longitude = longitude;
        this.Altitude = altitude;
        this.Address = address;
    }
}